src/systems/world_generation/WorldGeneration.ts

```diff
@@ -0,0 +1,171 @@
+import forest_path from "../../assets/environment/tiles/forest.png";
+import grass_path from "../../assets/environment/tiles/grass.png";
+import mountain_path from "../../assets/environment/tiles/mountains.png";
+import ocean_path from "../../assets/environment/tiles/ocean_deep.png";
+import { randomInt } from "../../helpers/math/MathHelper";
+
+// Define enum for tile types
+export enum TerrainType {
+    Forest,
+    Grass,
+    Mountain,
+    Ocean
```

Comment:
The typescript is pretty good! These should probably be an array of objects or record in some sense - so that we can extend them (eg. with stats), and wouldn't need to use TerrainType.name each call

I could help/think after work

Reply:
Do you mean like a predefined list of objects in constants that we pull from?